File: llmos-whiteboard/src/components/main_pages/prompt_mainboard/PromptWindows/types/WindowConfig.ts
import {WindowIcon, WindowTheme} from "../Theme";
import {ComponentType} from "react";

export interface WindowConfig {
  windowType: string;
  windowId: string;
  windowTitle: string;
  description?: string;
  windowTheme: WindowTheme;
  icon: WindowIcon;
  tabs: string[];
  [key: string]: any; // 允许后端扩展其他自定义字段
}

export interface WindowProps<T = {}> {
  data: any
  darkMode: boolean
  windowConfig: WindowConfig
  onUpdate?: any
  isUpdated?: boolean
  extra?: T  // 扩展字段
}

export type WindowComponent = ComponentType<WindowProps>

export interface RegistryEntry {
  component: WindowComponent; // 接受任何 React 组件
  defaultConfig: WindowConfig;
}

File: llmos_core/ui/windowConfig.py
from typing import Optional, Dict

from pydantic import BaseModel


class WindowConfig(BaseModel):
    """窗口配置模型"""
    windowId: str
    windowType: str
    windowTitle: str
    description: Optional[str] = None
    order: int = 0
    windowTheme: Optional[str] = None
    data:Dict[str,str] = {}